### ISSUE TITLE
GSConnect inhibits sleep while it's active

### ISSUE THREAD

**Issue by stel-bl** [GSConnect/gnome-shell-extension-gsconnect#1763]
### Describe the bug

The sleep is inhibited while GSConnect is active.

### Steps to reproduce

1. Install and enable GSConnect.
2. Try to suspend either by choosing "Suspend" from the power menu or `systemctl suspend`.

### Expected behavior

The system goes to sleep properly.

### GSConnect version

56

### Installed from

GNOME Extensions website

### GNOME Shell version

45.5

### Linux distribution/release

Arch

### Paired device(s)

Waydroid container with latest LineageOS 18.1 build, Redmi Note 10S running the latest version of official Evolution X

### KDE Connect app version

1.29.0

### Screenshots

![image](https://github.com/GSConnect/gnome-shell-extension-gsconnect/assets/39265461/e422397f-8707-4c46-b7c6-00f7b0d798e8)

### Notes

If I try to suspend by closing the lid, the screen turns off but the system doesn't suspend. Reopening the lid doesn't turn the screen back on until I forcefully shut the system down (probably GNOME or Nvidia problem).

Also I know this is GSConnect's doing because disabling it allows the system to suspend.

**Comment by andyholmes** [comment]
Thanks for reporting, this sounds like it's probably related to the [remote input override](https://github.com/GSConnect/gnome-shell-extension-gsconnect/blob/main/src/shell/input.js).

Have you tried disable that setting in the preferences?

**Comment by stel-bl** [comment]
> Thanks for reporting, this sounds like it's probably related to the [remote input override](https://github.com/GSConnect/gnome-shell-extension-gsconnect/blob/main/src/shell/input.js).
> 
> Have you tried disable that setting in the preferences?

I think I did and it didn't help. I can't test it anymore right now though because I already upgraded to GNOME 46 just a little while ago...

**Comment by andyholmes** [comment]
No worries, someone else can confirm if it's still a problem.

**Comment by stel-bl** [comment]
> No worries, someone else can confirm if it's still a problem.

Alrighty, thanks a lot!

**Comment by stel-bl** [comment]
It just hit my mind to try replicating this on a [Pardus](https://pardus.org.tr) 23.1 VM I had installed for a competition. The GNOME version of it has GNOME 43.9. I installed the extension from GNOME Extensions website using the preinstalled version of Firefox ESR and paired it with my phone, followed by an attempt to suspend using `systemctl suspend`. To my surprise, it did work fine and the VM could suspend properly. So yeah, I'm completely lost on what could be the cause now.

**Comment by andyholmes** [comment]
Hmm, well it looks like the override was first shipped with v55 for GNOME 45, so I guess that makes sense. Also it's another strong indication that it is the override preventing suspend.

Usually this is something you do for e.g. a text editor with an unsaved file, so when you attempt to suspend/shutdown there's a popup notifying you of the application inhibiting the session change.

I'm not sure that's a thing an extension can do, but then maybe the shell extension shouldn't be doing this but the service instead :thinking:

**Comment by stel-bl** [comment]
> Hmm, well it looks like the override was first shipped with v55 for GNOME 45, so I guess that makes sense. Also it's another strong indication that it is the override preventing suspend.

Glad to have clarified it. I guess I'll try to work around it when the GNOME 46 version is released.

> Usually this is something you do for e.g. a text editor with an unsaved file, so when you attempt to suspend/shutdown there's a popup notifying you of the application inhibiting the session change.
> 
> I'm not sure that's a thing an extension can do, but then maybe the shell extension shouldn't be doing this but the service instead 🤔

I guess there *is* a way to work around it, but you might not like my idea for this one.

Basically try to track user-initiated suspension triggers if possible (such as closed lid or `systemctl suspend` command). Once that's done, you should be able to make the extension disable its service on its own and start it back up when the computer wakes up.

Again, this is pretty much something I just came up with and might rather be a dirty hack instead of an actual fix.

> Thanks for checking back again, it's really helpful to get follow-ups :)

You're welcome! I've always had this issue in my watch list, so I just thought I could help narrow down the causes and try to further troubleshoot it.

**Comment by stel-bl** [comment]
Alright another followup.

After seeing the PR #1767, I decided to give it a shot and compile the extension with it, followed by an installation and whatnot.

I enabled GSConnect from the QS upon login, added the laptop's IP on my phone because I was connected to my phone's hotspot and... It works! I can finally suspend properly while GSConnect is active!

So I think GNOME 46 neutralized the way GSConnect handles remote input override...it maybe you guys updated the extension to not inhibit suspend. I have no idea how such advanced extensions work. :sweat_smile:

But yeah, I think it's safe to close this issue, and let others know that it's been fixed *at least with GNOME 46*.

Good luck on the development!
